x-pack/platform/packages/shared/onechat/onechat-genai-utils/tools/generate_esql/prompts/instructions_template.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export interface InstructionsTemplateParams {
+  /**
+   * The default LIMIT to use when no specific limit is requested by the user.
+   */
+  defaultLimit?: number;
+  /**
+   * The maximum LIMIT to use when the user asks for "all" results.
+   */
+  maxAllLimit?: number;
+}
+
+/**
+ * Generates ES|QL query generation instructions with configurable limit values.
+ * This is a copy of the instructions from the inference plugin, modified to support
+ * custom row limits for Agent Builder's index search tool.
+ */
+export const getEsqlInstructions = (params: InstructionsTemplateParams = {}): string => {
+  const { defaultLimit = 100, maxAllLimit = 250 } = params;
+
+  return `<instructions>
```

Comment:
Copied from and made it a template https://github.com/elastic/kibana/blob/c495727dd32fbbbe5b180f7cb2ce89588058799b/x-pack/platform/plugins/shared/inference/server/tasks/nl_to_esql/prompts/instructions.txt#L17-L25